### PR TITLE
[Draft] py-geant4 (g4py)

### PR DESCRIPTION
**Pull request by ax3l**
Draft of a geant4 binding for python.

Same-version-pinning as already working in `adios` package for a very similar situation.

cc @drbenmorgan can you help? :)

cc @LDAmorim

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-geant4/package.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+from spack import *
+
+
+class PyGeant4(CMakePackage):
+    """Python bindings for Geant4 (g4py)"""
+
+    homepage = "http://geant4.cern.ch/"
+    git = "https://gitlab.cern.ch/geant4/geant4.git"
+    url = "https://gitlab.cern.ch/geant4/geant4/-/archive/v10.6.0/geant4-v10.6.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-geant4/package.py]:
```suggestion
    url = "https://gitlab.cern.ch/geant4/geant4/-/archive/v10.6.1/geant4-v10.6.1.tar.gz"
```
It doesn't make a difference for fetching, but our URL accuracy is based on how many versions that we extract from a URL are actually listed in the package.

File: var/spack/repos/builtin/packages/py-geant4/package.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    depends_on('cmake@3.3:', type='build')
+
+    # C++11/14/17 support
+    depends_on('xerces-c cxxstd=11', when='cxxstd=11')
+    depends_on('xerces-c cxxstd=14', when='cxxstd=14')
+    depends_on('xerces-c cxxstd=17', when='cxxstd=17')
+
+    depends_on('boost +python')
+    depends_on('python')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-geant4/package.py]: Where do the Python libraries get installed to? If they get installed to `lib/pythonX.Y/site-packages`, then this should be:
```suggestion
    extends('python')
```
If not, you'll need to set up `PYTHONPATH` manually.

**Comment by drbenmorgan** [comment]
Geant4's recently completed integration of Geant4Py with the main Geant4 build. That'll be out in June as a patch to 10.6 and preview in the 10.7 beta.

I'd recommend waiting for those to be out, then the core geant4 spackage can have a `python` variant added for 10.6.2 and above. These will be Python3 exclusive and will require Boost.Python still. Migration to pybind11 was attempted, but is going to be a bit more than `sed`ing Boost.Python to pybind11. That's going to have to wait until the next release.

**Comment by drbenmorgan** [comment]
Geant4 10.6.2 with Python support is now out - I'll prepare a PR with that in plus Python support (so new +python variant).

**Comment by ax3l** [comment]
Thank you so much, I'll give it a try!
Please don't hesitate to reach out to the pybind11 community, most active in issues on github and on https://gitter.im/pybind/Lobby, in case you need help.

**Comment by ax3l** [comment]
Closed in favor of #16868

**Pull request closed without merging**